src/System.Numerics.Matrices/src/System/Numerics/Matrix5x5.cs

```diff
@@ -0,0 +1,543 @@
+        /// </summary>
+        public int Columns { get { return ColumnCount; } }
+        /// <summary>
+        /// Get the number of rows in the matrix
+        /// </summary>
+        public int Rows { get { return RowCount; } }
+
+        /// <summary>
+        /// Gets a new Matrix1x5 containing the values of column 1
+        /// </summary>
+        public Matrix1x5 Column1 { get { return new Matrix1x5(M11, M12, M13, M14, M15); } }
+        /// <summary>
+        /// Gets a new Matrix1x5 containing the values of column 2
+        /// </summary>
+        public Matrix1x5 Column2 { get { return new Matrix1x5(M21, M22, M23, M24, M25); } }
+        /// <summary>
+        /// Gets a new Matrix1x5 containing the values of column 3
+        /// </summary>
+        public Matrix1x5 Column3 { get { return new Matrix1x5(M31, M32, M33, M34, M35); } }
+        /// <summary>
+        /// Gets a new Matrix1x5 containing the values of column 4
+        /// </summary>
+        public Matrix1x5 Column4 { get { return new Matrix1x5(M41, M42, M43, M44, M45); } }
+        /// <summary>
+        /// Gets a new Matrix1x5 containing the values of column 5
+        /// </summary>
+        public Matrix1x5 Column5 { get { return new Matrix1x5(M51, M52, M53, M54, M55); } }
+        /// <summary>
+        /// Gets a new Matrix5x1 containing the values of column 1
+        /// </summary>
+        public Matrix5x1 Row1 { get { return new Matrix5x1(M11, M21, M31, M41, M51); } }
+        /// <summary>
+        /// Gets a new Matrix5x1 containing the values of column 2
+        /// </summary>
+        public Matrix5x1 Row2 { get { return new Matrix5x1(M12, M22, M32, M42, M52); } }
+        /// <summary>
+        /// Gets a new Matrix5x1 containing the values of column 3
+        /// </summary>
+        public Matrix5x1 Row3 { get { return new Matrix5x1(M13, M23, M33, M43, M53); } }
+        /// <summary>
+        /// Gets a new Matrix5x1 containing the values of column 4
+        /// </summary>
+        public Matrix5x1 Row4 { get { return new Matrix5x1(M14, M24, M34, M44, M54); } }
+        /// <summary>
+        /// Gets a new Matrix5x1 containing the values of column 5
+        /// </summary>
+        public Matrix5x1 Row5 { get { return new Matrix5x1(M15, M25, M35, M45, M55); } }
+
+        public override bool Equals(object obj)
+        {
+            if (obj is Matrix5x5)
+                return this == (Matrix5x5)obj;
+
+            return false;
+        }
+
+        public bool Equals(Matrix5x5 other)
+        {
+            return this == other;
+        }
+
+        public unsafe override int GetHashCode()
+        {
+            fixed (Matrix5x5* p = &this)
+            {
+                int* x = (int*)p;
+                unchecked
+                {
+                    return 0xFFE1
+                         + 7 * ((((x[00] ^ x[01]) << 0) + ((x[02] ^ x[03]) << 1) + ((x[04] ^ x[05]) << 2) + ((x[06] ^ x[07]) << 3) + ((x[08] ^ x[09]) << 4)) << 0)
```

Comment:
This logic could be hokum, somebody with knowledge of hashing should take a closer look. This kind of logic has always worked finely for me in the past, but it very well could be over designed and/or a terrible algo.